test/test_client_async_asyncio.py

```diff
@@ -239,7 +242,7 @@ def testClientProtocolConnectionLost(self, protocol):
             assert protocol.factory.protocol_lost_connection.call_count == 1
 
     @pytest.mark.parametrize("protocol", protocols)
-    def testClientProtocolDataReceived(self, protocol):
+    async def test_client_protocol_data_received(self, protocol):
```

Comment:
I am not into the code deeply atm.
Why does this method (and the other changes to async) not containing an await?
Does it need to be async then? Or did I missed sth?

Reply:
Await can only be used in async functions, but are not needed to be used. Pytest detects it is an async function and does await internally.

In general all functions dealing with async (e.g. calling async functions or yielding) should be declared as async, this replaces the old coroutine decorator.

Reply:
with 3.10 you can experience function hanging if not declared correctly.